src/zod-extensions.ts

```diff
@@ -48,13 +48,15 @@ export function extendZodWithOpenApi(zod: typeof z) {
     });
 
     if (isZodType(this, 'ZodObject')) {
-      const initialExtend = this.extend;
+      const originalExtend = this.extend;
 
       result.extend = function (...args: any) {
-        const extendedResult = initialExtend.apply(result, args);
+        const extendedResult = originalExtend.apply(this, args);
 
         extendedResult._def.openapi = {
-          extendedFrom: result._def.openapi?.refId,
+          extendedFrom: this._def.openapi?.refId
+            ? { refId: this._def.openapi?.refId, schema: this }
```

Comment:
You can remove the optional chaining here, right?

Reply:
Done